bitcoin/src/crypto/taproot.rs

```diff
@@ -26,23 +26,21 @@ pub struct Signature {
 
 impl Signature {
     /// Deserialize from slice
-    pub fn from_slice(sl: &[u8]) -> Result<Self, Error> {
+    pub fn from_slice(sl: &[u8]) -> Result<Self, SigFromSliceError> {
         match sl.len() {
             64 => {
                 // default type
-                let sig =
-                    secp256k1::schnorr::Signature::from_slice(sl)?;
+                let sig = secp256k1::schnorr::Signature::from_slice(sl)?;
                 Ok(Signature { sig, hash_ty: TapSighashType::Default })
             }
             65 => {
                 let (hash_ty, sig) = sl.split_last().expect("Slice len checked == 65");
                 let hash_ty = TapSighashType::from_consensus_u8(*hash_ty)
-                    .map_err(|_| Error::InvalidSighashType(*hash_ty))?;
-                let sig =
-                    secp256k1::schnorr::Signature::from_slice(sig)?;
+                    .map_err(|_| SigFromSliceError::InvalidSighashType(*hash_ty))?;
```

Comment:
Maybe there should be `From` conversion?

Reply:
`sighash::Error` needs splitting up, its on my todo list. Then the `SigFromSliceErorr` can have a variant that holds the error returned here.

In case you are interested, I audited all errors yesterday and there are actually not that many that need splitting up, we are close to having good errors I believe. My list of modules that contain errors that need improving is:

- bip158
- bip32
- script/mod
- crypto/ecdsa
- crypto/key
- crypto/sighash